ocs_ci/ocs/bucket_utils.py

```diff
@@ -2554,6 +2554,48 @@ def get_object_count_in_bucket(io_pod, bucket_name, prefix="", s3_obj=None):
     return len(output.splitlines())
 
 
+def wait_for_bucket_count_stability(
+    mcg_obj, expected_count=None, max_retries=10, retry_interval=5, stability_count=3
+):
+    """
+    Poll CLI until bucket count is stable or expected count is reached.
+
+    Args:
+        mcg_obj: MCG object instance with cli_list_all_buckets method
+        expected_count (int, optional): Expected number of buckets
+        max_retries (int): Maximum number of retry attempts
+        retry_interval (int): Seconds between retries
+        stability_count (int): Number of consecutive identical counts to consider stable
+
+    Returns:
+        tuple: (final_count, is_expected_reached)
+    """
+    previous_counts = []
+    for attempt in range(max_retries):
+        cli_buckets = mcg_obj.cli_list_all_buckets()
+        current_count = len(cli_buckets)
+
+        logger.info(
+            f"Bucket count from CLI (attempt {attempt+1}/{max_retries}): {current_count}"
```

Comment:
f"Bucket count from CLI: {current_count}. Attempt {attempt+1} out of {max_retries})"

Reply:
fixed